File: Codigos estudiantes por lenguaje/Java/Edwin Villarraga/binarySearch.java
public class Main {
	static int binarySearch(int arr[], int l, int r, int x) {
		if (r >= l) {
			int mid = l + (r - l) / 2;
			if (arr[mid] == x) {
				return mid;
			}
			return binarySearch(arr, mid + 1, r, x);
		}
		return -1;
	}

	public static void main(String args[]) {
		int arr[] = { 2, 3, 4, 10, 40 };
		int n = arr.length;
		int x = 10;
		int result = binarySearch(arr, 0, n - 1, x);
		if (result == -1) {
			System.out.println("Elemento no presente");
		} else {
			System.out.println("Elemento fue encontrado en la posición " + result);
		}
	}
}

File: Codigos estudiantes por lenguaje/C++/Diana Molares/introduccion/binarysearch.cpp
#include <bits/stdc++.h>
#include <cstdlib>

static int binarysearch(int arr[],int l,int r,int x){
	if(r>=l){
		int mid=l+(r-1)/2;
			if(arr[mid]==x){
		return mid;
	}
	if(arr[mid]>x){
		return binarysearch(arr,l,mid-1,x);
	}else{
		return binarysearch(arr,mid+1,r,x);
	}
	}

	return -1;
}
int main() {
	int arr[]={2,3,4,5,6};
	int n=sizeof(arr)/sizeof(arr[0]);
	int x=5;
	int resultado=binarysearch(arr,0,n-1,x);
	if(resultado==-1){
		printf("elemento no presente");
	}else{
		printf("elemento encontrado en el indice: %d",resultado);
	}
	return 0;
}
